subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -325,8 +317,6 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv)
 	if (atomic_test_bit(ext_adv->flags, ADV_FLAG_ACTIVE)) {
 		atomic_set_bit(ext_adv->flags, ADV_FLAG_SCHEDULE_PENDING);
 		return false;
-	} else if (atomic_test_and_set_bit(ext_adv->flags, ADV_FLAG_SCHEDULED)) {
-		return false;
```

Comment:
Have you considered that `k_work_submit` is not executed immediately, but in an asynchronous manner.

Assuming that there are multiple mesh messages that need to be relayed in an advertising event report(it is possible) then after your change this time, only one ext adv will be scheduled repeatedly. The default method was that multiple relay messages would occupy different ext advs respectively.

Reply:
Right, I missed that it returns `false` here. Added `k_work_is_pending`.